src/error.rs

```diff
@@ -0,0 +1,44 @@
+//!
+//! Wrapper for xmlError
+//!
+use super::bindings;
+
+use std::ffi::CStr;
+
+
+/// Wrapper around xmlErrorPtr
+#[derive(Debug)]
+pub struct XmlStructuredError(*mut bindings::_xmlError);
```

Comment:
I am not familiar enough with the approach of "structured errors", which is my only hesitation in merging right away. Are these worth adopting for all errors produced by libxml2, or are they specific to the schema reporting? If the latter, they ought to be under `libxml::schema::error`, and if the former, I would need to think about which of the other error interfaces should be refactored to use these - and it they're the right abstraction.

You're also using a FFI-like name with `XmlStructuredError`, which is something nicely compartmentalized in `bindings.rs` -- all "higher" modules stay away from that naming convention. If these were to be the main error interface of the crate, simply naming the struct `Error` would be completely acceptable.

Reply:
They represent a generalized way of reporting errors in libxml2. But to be frank I myself find the whole error handling thing in libxml2 a little confusing and difficult to work with. I get the distinct feeling, that it also grew organically out of the necessities of each module being implemented along the way. 

Perhaps we should dig deeper and come up with the cleanest unified approach possible.

As for changing the name, sure thing. No problem. On the other hand I'd prefer not to go for "Error" for it being to generic. Would "StructuredError" be ok? Error is something more reserved to error handling the Rust way, which is not the case unless we make it so.

Reply:
One more thing; would you be opposed to me opening an issue for discussion on the level of abstraction being aimed at in this library wrapper? I'm currently having trouble getting a clear distinction between high and low levels and what they are. It feels a little awkward. Sometimes the wrapping is high level sometimes low. Perhaps it would be sensible to split the project into two layers?

Reply:
Indeed, I was just discussing the low vs high levels in #60 , and we may indeed benefit from a discussion and a convention. This wrapper is also growing organically, and is starting to try and service several competing use patterns, so it may be a good time to take a step back and discuss.

Reply:
Also `StructuredError` is a less generic name than `Error`, but it is also a somewhat poor name from the libxml2 folks, more confusing than informative.

It is also available in bindgen.rs even in master, so I am unsure what benefit there is to draw from recreating it as part of the wrapper interfaces - might as well unify the error-handling for the whole crate before we move forward here.

Reply:
> Indeed, I was just discussing the low vs high levels in #60 , and we may indeed benefit from a discussion and a convention. This wrapper is also growing organically, and is starting to try and service several competing use patterns, so it may be a good time to take a step back and discuss.

Ok, I'll draft a proposal for discussion in a new issue. Will take some time, since I do not want to jump into discussion without having a clear picture on the topic myself.

> Also `StructuredError` is a less generic name than `Error`, but it is also a somewhat poor name from the libxml2 folks, more confusing than informative.

Agreed.

> It is also available in [bindings.rs](https://github.com/KWARC/rust-libxml/blob/master/src/bindings.rs#L6023) even in master, so I am unsure what benefit there is to draw from recreating it as part of the wrapper interfaces - might as well unify the error-handling for the whole crate before we move forward here.

You are right. Clarity in the way forward might help to answer the question whether it is a useful approach or not.